Cairo Desktop/Cairo Desktop/Taskbar.xaml.cs

```diff
@@ -141,21 +141,21 @@ private void setTaskbarDesktopOverlayButton()
 
         private void setTaskbarSize()
         {
-            switch ((IconSize)Settings.Instance.TaskbarIconSize)
+            switch ((IconSize.Sizes)Settings.Instance.TaskbarIconSize)
             {
-                case IconSize.Large:
+                case IconSize.Sizes.Large:
                     addToSize = 16;
                     break;
-                case IconSize.Medium:
+                case IconSize.Sizes.Medium:
                     addToSize = 8;
                     break;
                 default:
                     addToSize = 0;
                     break;
             }
 
-            baseButtonWidth = 140 + addToSize;
-            Height = 29 + addToSize;
+            baseButtonWidth = Settings.Instance.HideTaskbarLabels ? getButtonWidth() : (Settings.Instance.TaskbarButtonWidth + addToSize);
```

Comment:
For setting the width of the taskbar buttons when labels are hidden, I would suggest simply using the height so that each button is square. This also has the benefit of eliminating the need for `getButtonWidth()`. For example, this line could change to: `baseButtonWidth = (Settings.Instance.HideTaskbarLabels ? Settings.Instance.TaskbarButtonHeight : Settings.Instance.TaskbarButtonWidth) + addToSize;`

Reply:
Indeed, that function was useless and a perfect square is the way to go.